api/python/provisioner/errors.py

```diff
@@ -80,9 +108,14 @@ def __init__(
         self.reason = reason
 
     def __str__(self):
+        logger.error(
```

Comment:
@mazinamdar that is too far, please don't do that in such a way: `__str__` should just return string.

Reply:
As an option:
- leave that `__str__` as is
- call logging for the error explicitly in the place where it is raised (we don't have a single place to deal with that for now) with a wrapped output so two log lines would appear: wrapped one first, legacy one-lined - second.

It's a very dummy/dirty option as well but better than current one.

Better option would be:
- keep that one-line but log only valuable data (e.g. do not log the states that are not failed)

Reply:
For now added error log where we are raising exceptions. In case of failure I think it is better to display complete information to user so that it can help to debug the issue.